packages/web/src/common/store/tipping/sagas.ts

```diff
@@ -322,90 +319,86 @@ function* refreshSupportAsync({
   payload: RefreshSupportPayloadAction
   type: string
 }) {
-  const audiusBackendInstance = yield* getContext('audiusBackendInstance')
-  const encodedSenderUserId = encodeHashId(senderUserId)
-  const encodedReceiverUserId = encodeHashId(receiverUserId)
-
-  if (encodedSenderUserId && encodedReceiverUserId) {
-    const supportingParams: SupportRequest = {
-      encodedUserId: encodedSenderUserId,
-      audiusBackendInstance
-    }
-    if (supportingLimit) {
-      supportingParams.limit = supportingLimit
-    } else {
-      yield* waitForAccount()
-      const account = yield* select(getAccountUser)
-      supportingParams.limit =
-        account?.user_id === senderUserId
-          ? account?.supporting_count
-          : MAX_ARTIST_HOVER_TOP_SUPPORTING + 1
-    }
+  const apiClient = yield* getContext('apiClient')
 
-    const supportersParams: SupportRequest = {
-      encodedUserId: encodedReceiverUserId,
-      audiusBackendInstance
-    }
-    if (supportersLimit) {
-      supportersParams.limit = supportersLimit
-    }
+  const supportingParams: GetSupportingArgs = {
+    userId: senderUserId
+  }
+  if (supportingLimit) {
+    supportingParams.limit = supportingLimit
+  } else {
+    yield* waitForAccount()
+    const account = yield* select(getAccountUser)
+    supportingParams.limit =
+      account?.user_id === senderUserId
+        ? account?.supporting_count
+        : MAX_ARTIST_HOVER_TOP_SUPPORTING + 1
+  }
 
-    const supportingForSenderList = yield* call(
-      fetchSupporting,
-      supportingParams
-    )
-    const supportersForReceiverList = yield* call(
-      fetchSupporters,
-      supportersParams
-    )
+  const supportersParams: GetSupportersArgs = {
+    userId: receiverUserId
+  }
+  if (supportersLimit) {
+    supportersParams.limit = supportersLimit
+  }
 
-    const userIds = [
-      ...supportingForSenderList.map((supporting) =>
-        decodeHashId(supporting.receiver.id)
-      ),
-      ...supportersForReceiverList.map((supporter) =>
-        decodeHashId(supporter.sender.id)
-      )
-    ]
+  const supportingForSenderList = yield* call(
+    [apiClient, apiClient.getSupporting],
+    supportingParams
+  )
+  const supportersForReceiverList = yield* call(
+    [apiClient, apiClient.getSupporters],
+    supportersParams
+  )
 
-    yield call(fetchUsers, userIds)
+  if (!supportingForSenderList || !supportersForReceiverList) return
```

Comment:
this makes sense, but want to confirm that early exit is all good

Reply:
yep should be, but will wait for @sddioulde too. we try/caught things before and just didn't dispatch later actions

Reply:
hmm im not sure this early return is correct. it's saying if there is either no supporting for sender or no supporters for receiver then return. What if there is no supporting for sender but there are supporters for the receiver?

Reply:
I really don't think this is an issue because apiclient will only return a nullable response if the request errors out. Otherwise response.data = [] and this will still be truthy.

I switched to an inline ternary just in case, but I think this is good.